modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js

```diff
@@ -138,36 +168,41 @@ function FieldBase({
 					</div>
 				)}
 
-				{((label && showLabel) ||
-					required ||
-					tooltip ||
-					repeatable) && (
-					<label
-						className={classNames({
-							'ddm-empty': !showLabel && !required,
-							'ddm-label': showLabel || required,
-						})}
-						htmlFor={id ? id : name}
-					>
-						{label && showLabel && label}
-
-						{required && (
-							<span className="reference-mark">
-								<ClayIcon symbol="asterisk" />
-							</span>
-						)}
-
-						{tooltip && (
-							<span className="ddm-tooltip">
-								<ClayIcon
-									data-tooltip-align="right"
-									symbol="question-circle-full"
-									title={tooltip}
-								/>
-							</span>
-						)}
-					</label>
-				)}
+				{showLegend &&
+					((label && showLabel) ||
+						required ||
+						tooltip ||
+						repeatable) && (
+						<legend className="lfr-ddm-legend">
+							{label && showLabel && label}
+
+							<FieldProperties
+								required={required}
+								tooltip={tooltip}
+							/>
+						</legend>
+					)}
+
+				{!showLegend &&
+					((label && showLabel) ||
+						required ||
+						tooltip ||
+						repeatable) && (
+						<label
+							className={classNames({
+								'ddm-empty': !showLabel && !required,
+								'ddm-label': showLabel || required,
+							})}
+							htmlFor={id ? id : name}
+						>
+							{label && showLabel && label}
+
```

Comment:
@carolmariaabb, why are you checking the `label` two times on this `if`? Is this necessary?

Reply:
Hi @natocesarrego, this was actually added by this commit: [Migrate FieldBase to React and add Adapter for compatibility](https://github.com/liferay/liferay-portal/commit/b18a778cec5a2145fd7d0341a60360e1cf1447d4).